monkeytype/tracing.py

```diff
@@ -215,7 +215,7 @@ def handle_call(self, frame: FrameType) -> None:
         if frame in self.traces:
             # resuming a generator; we've already seen this frame
             return
-        arg_names = code.co_varnames[0 : code.co_argcount]
+        arg_names = code.co_varnames[: code.co_argcount + code.co_kwonlyargcount]
```

Comment:
It seems this can also support `*args` and `**kwargs` by doing
```suggestion
        arg_names = code.co_varnames
```

Reply:
I had that idea too but this alone doesn't work. The code that generates the annotation for the `**kwargs` needs to be adapted as well. The problem is that if a function `foo(**kwargs)` is called as `foo(a=1)`, the correct annotation for `kwargs` would be `int` instead of `dict[str, int]`, which is what happens without additional changes. E.g.:

```python
def uses_kwargs(arg, **kwargs):
    return arg + len(kwargs)

uses_kwargs(1, a=1)
uses_kwargs(1, b=2)
uses_kwargs(1, c='', d='')
```

This will generate the following stub, which doesn't work:

```python
def uses_kwargs(arg: int, **kwargs: Dict[str, Union[str, int]]) -> int: ...
```